src/applications/bmqstoragetool/m_bmqstoragetool_searchresultfactory.cpp

```diff
@@ -0,0 +1,231 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// m_bmqstoragetool_searchresultfactory.cpp -*-C++-*-
+
+// bmqstoragetool
+#include <m_bmqstoragetool_searchresultfactory.h>
+
+namespace BloombergLP {
+namespace m_bmqstoragetool {
+
+namespace {
+
+// Helpers to create base searchResult implementation
+void createSearchDetailResult(bsl::shared_ptr<SearchResult>&  baseResult,
+                              bsl::ostream&                   ostream,
+                              const QueueMap&                 queueMap,
+                              bsl::shared_ptr<PayloadDumper>& payloadDumper,
+                              bslma::Allocator*               allocator,
+                              bool                            printImmediately,
+                              bool                            eraseDeleted,
+                              bool                            cleanUnprinted)
+{
+    baseResult.reset(new (*allocator) SearchDetailResult(ostream,
+                                                         queueMap,
+                                                         payloadDumper,
+                                                         allocator,
+                                                         printImmediately,
+                                                         eraseDeleted,
+                                                         cleanUnprinted),
+                     allocator);
+}
+
+void createSearchShortResult(bsl::shared_ptr<SearchResult>&  baseResult,
+                             bsl::ostream&                   ostream,
+                             bsl::shared_ptr<PayloadDumper>& payloadDumper,
+                             bslma::Allocator*               allocator,
+                             bool                            printImmediately,
+                             bool                            printOnDelete,
+                             bool                            eraseDeleted)
+{
+    baseResult.reset(new (*allocator) SearchShortResult(ostream,
+                                                        payloadDumper,
+                                                        allocator,
+                                                        printImmediately,
+                                                        printOnDelete,
+                                                        eraseDeleted),
+                     allocator);
+}
```

Comment:
What is the purpose of this helpers? They only forward the parameters. The code, calling them isn't shorter than their implementation. So in my opinion, these fucntuions only confuse

Reply:
Removed helpers